railties/lib/rails/generators/app_base.rb

```diff
@@ -298,6 +298,40 @@ def rails_version_specifier(gem_version = Rails.gem_version)
         end
       end
 
+      # This "npm-ifies" the current version number
+      # With npm, versions such as "5.0.0.rc1" or "5.0.0.beta1.1" are not compliant with its
+      # versioning system, so they must be transformed to "5.0.0-rc1" and "5.0.0-beta1-1" respectively.
+
+      # "5.0.1"     --> "5.0.1"
+      # "5.0.1.1"   --> "5.0.1-1" *
+      # "5.0.0.rc1" --> "5.0.0-rc1"
+      #
+      # * This makes it a prerelease. That's bad, but we haven't come up with
+      # a better solution at the moment.
+      def npm_version
+        # TODO: support `options.dev?`
+
+        if options.edge? || options.main?
+          # TODO: ideally this would read from Github
+          # https://github.com/rails/rails/blob/main/actioncable/app/assets/javascripts/action_cable.js
+          # https://github.com/rails/rails/blob/main/activestorage/app/assets/javascripts/activestorage.js
```

Comment:
Because https://github.com/rails/rails/blob/main/activestorage/package.json isn't a top level file - unlike https://github.com/turbolinks/turbolinks, or the examples in https://docs.npmjs.com/cli/v7/configuring-npm/package-json#git-urls-as-dependencies - I'm not sure how to point a gif ref to it.

Reply:
`latest` is a bit of a compromise. I imagine someone who knows more about npm than me could make this work.